R/submit_ipr.R

```diff
@@ -1,145 +1,417 @@
+    idx_start <- nrow(tb_joined)
+    cat("Joined", input_file, "to final table\n")
+  }
+  tb_joined <- tibble::as_tibble(tb_joined)
+
+  cat("Writing joined table to", outfolder, "\n")
+  write.table(tb_joined, file = file.path(outfolder, "ipr_joined.tsv"), sep = "\t", quote = FALSE, row.names = FALSE)
+  return(tb_joined)
 }
 
-submit_ipr <- function(path2seq, outfolder, email) {
-#' @export
-#' @title Submit IPRscan analysis for protein sequences (multifasta input)
-#' @param path2seq path to your sequence file
-#' @param outfolder location for IPRscan outputs
-#' @param email required email ID for job submission (to track and prevent accidental misuse of the API)
-#' @keywords domains, domain architectures, protein characterization
+submit_ipr <- function(path2seq, outfolder, user_email) {
+  #' @export
+  #' @title Submit IPRscan analysis for protein sequences (multifasta input)
+  #' @param path2seq path to your sequence file
+  #' @param outfolder location for IPRscan outputs
+  #' @param email required email ID for job submission (to track and prevent accidental misuse of the API)
+  #' @keywords domains, domain architectures, protein characterization
 
+  total_jobs <<- 0
   outfolder <- file.path(outfolder)
   if (!(dir.exists(outfolder))) {
     dir.create(outfolder)
   }
+  # copy input file to the output folder
+  file.copy(from = path2seq, to = file.path(outfolder, "input.fa"))
+
+  tb_status <<- .construct_status_table()
 
-  # test if input is a multifasta file
   n_seqs <- nrow(Biostrings::fasta.index(path2seq))
 
+  ### single seq submit
   if (n_seqs == 1L) {
-    cat("Single sequence detected", "\n", sep = "")
-    outfile <- file.path(outfolder, "iprout")
-    .submit(path2seq, outfile, email)
-    cat("Full submission complete. Results located at: ",
-        outfolder, "\n", sep = "")
-    return(outfolder)
+    cat("Single sequence detected\n")
+    outfile <- file.path(outfolder, "iprout.tsv")
+
+    # handle POST fail
+    submission_outcome <- .submit(path2seq, user_email)
+    # .submit returns either "failed" or a job_id
+    job_id <- ifelse(submission_outcome != "failed", submission_outcome, NULL)
+    if (submission_outcome == "failed") {
+      message <- cat("Single submission failed\n")
+      warning(message)
+      return(NULL)
+    }
+    # cat("Full submission complete. Results located at: ", outfolder, "\n", sep = "")
+    status <- "submitted"
+    # continue while status does not equal 'complete' or 'failed'
+
+    n_polls <- 1L
+    while (!(status == "completed" | status == "failed")) {
+      status <- .poll_job(job_id)
+      .update_status_table(job_id, status)
+      # quit after 180 tries; 15mins
+      if (n_polls > 180) {
+        status <- "failed"
+      }
+      n_polls <- n_polls + 1
+      Sys.sleep(5)
+    }
+
+    if (status == "completed") {
+      tb_result <- .get_job_result(job_id, outfolder, write_result = FALSE)
+      cols <- c(
```

Comment:
can be a separate function (it's in a different script); that way @falquaddoomi can call it too. We will have one with original and one with cleaned up names.